pkg/cmdutil/json_flags.go

```diff
@@ -259,3 +263,35 @@ type exportable interface {
 var exportableType = reflect.TypeOf((*exportable)(nil)).Elem()
 var sliceOfEmptyInterface []interface{}
 var emptyInterfaceType = reflect.TypeOf(sliceOfEmptyInterface).Elem()
+
+// Basic function that can be used with structs that need to implement
+// the exportable interface. It has numerous limitations so verify
+// that it works as expected with the struct and fields you want to export.
+// If it does not, then implementing a custom ExportData method is necessary.
+// Perhaps this should be moved up into exportData for the case when
+// a struct does not implement the exportable interface, but for now it will
+// need to be explicitly used.
+func StructExportData(s interface{}, fields []string) map[string]interface{} {
```

Comment:
I would love to see all these export functions moved into their own internal function named `export` or `exportable` one day, but that felt like too big of a refactor to tack onto this PR.

Reply:
Additionally this name is not the greatest, open to other suggestions.